### PR TITLE
accounts/abi/bind: add void if no return args specified

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
Currently the java generator generates invalid input on pure/view functions
that have no return type. e.g. `function f(uint u) view public {}`
This is not a problem in practice as people rarely ever write functions like this.

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
Generated before this:
```
public  f(CallOpts opts, BigInt u) throws Exception { /* // TODO void not correctly set
		Interfaces args = Geth.newInterfaces(1);
		Interface arg0 = Geth.newInterface();arg0.setBigInt(u);args.set(0,arg0);
		Interfaces results = Geth.newInterfaces(0);	

		if (opts == null) {
			opts = Geth.newCallOpts();
		}
		this.Contract.call(opts, results, "f", args);	
	}
```
After:
```
public void f(CallOpts opts, BigInt u) throws Exception { /* // TODO void not correctly set
		Interfaces args = Geth.newInterfaces(1);
		Interface arg0 = Geth.newInterface();arg0.setBigInt(u);args.set(0,arg0);
		Interfaces results = Geth.newInterfaces(0);	

		if (opts == null) {
			opts = Geth.newCallOpts();
		}
		this.Contract.call(opts, results, "f", args);	
	}
```

File: accounts/abi/bind/template.go [hunk 1]
```diff
@@ -625,7 +625,7 @@ import java.util.*;
 	// {{.Normalized.Name}} is a free data retrieval call binding the contract method 0x{{printf "%x" .Original.ID}}.
 	//
 	// Solidity: {{.Original.String}}
-	public {{if gt (len .Normalized.Outputs) 1}}{{capitalise .Normalized.Name}}Results{{else}}{{range .Normalized.Outputs}}{{bindtype .Type $structs}}{{end}}{{end}} {{.Normalized.Name}}(CallOpts opts{{range .Normalized.Inputs}}, {{bindtype .Type $structs}} {{.Name}}{{end}}) throws Exception {
+	public {{if gt (len .Normalized.Outputs) 1}}{{capitalise .Normalized.Name}}Results{{else}}{{if eq (len .Normalized.Outputs) 0}}void{{else}}{{range .Normalized.Outputs}}{{bindtype .Type $structs}}{{end}}{{end}}{{end}} {{.Normalized.Name}}(CallOpts opts{{range .Normalized.Inputs}}, {{bindtype .Type $structs}} {{.Name}}{{end}}) throws Exception {
```
Comment (rjl493456442) [inline accounts/abi/bind/template.go:628]: This equals to 

```go
if len(output) > 1  {
  
} else {
  if len(output) == 0 {
    // void
  } else {

  }
}
```

We should use `public {{if gt (len .Normalized.Outputs) 1}}{{capitalise .Normalized.Name}}Results{{else if }} {{else}} {{end}}`

Comment (MariusVanDerWijden) [reply accounts/abi/bind/template.go:628]: Ahh I didn't know there was `else if` :D Is there a documentation for this language?

Comment (rjl493456442) [reply accounts/abi/bind/template.go]: https://golang.org/pkg/text/template/#hdr-Actions

**Pull request merged**